tests/test_denest.py

```diff
@@ -182,89 +179,104 @@ def test__schema__nested_arrays_add_tables():
-        [],
-        [{"a": {"b": []}},
-         {"a": {"b": [{"c": {"d": 1}}]}},
-         {"a": {"b": [{"c": {"d": 12}},
-                      {"c": {"d": 123}}]}},
-         {"a": {"b": [{"c": {"d": 1234}},
-                      {"c": {"d": 12345}},
-                      {"c": {"d": 123456}}]}},
-         {"a": {"b": [{"c": {"e": [{"f": "hello",
-                                    "g": True},
-                                   {"f": "goodbye",
-                                    "g": True}]}}]}}])
+NESTED_SCHEMA = {
+    "properties": {
+        "a": {"type": "object",
+              "properties": {
+                  "b": {
+                      "type": "array",
+                      "items": {
+                          "type": "object",
+                          "properties": {
+                              "c": {
+                                  "type": "object",
+                                  "properties": {
+                                      "d": {"type": "integer"},
+                                      "e": {"type": "array",
+                                            "items": {"type": "object",
+                                                      "properties": {
+                                                          "f": {"type": "string"},
+                                                          "g": {"type": "boolean"}}}}}}}}}}}}}
+
+NESTED_RECORDS = [{"a": {"b": []}},
+                  {"a": {"b": [{"c": {"d": 1}}]}},
+                  {"a": {"b": [{"c": {"d": 12}},
+                               {"c": {"d": 123}}]}},
+                  {"a": {"b": [{"c": {"d": 1234}},
+                               {"c": {"d": 12345}},
+                               {"c": {"d": 123456}}]}},
+                  {"a": {"b": [{"c": {"e": [{"f": "hello",
+                                             "g": True},
+                                            {"f": "goodbye",
+                                             "g": True}]}}]}}]
+
+
+def test__records__nested__tables():
+    denested = denest.to_table_batches(NESTED_SCHEMA, [], NESTED_RECORDS)
 
     print('denested:', denested)
 
     assert 3 == len(denested)
 
-    root_table_checked = False
-    nested_table_checked = False
-    nested_nested_table_checked = False
-
     for table_batch in denested:
-        if tuple() == table_batch['streamed_schema']['path']:
-            root_table_checked = True
+        assert table_batch['streamed_schema']['path'] in \
+               {tuple(),
+                ('a', 'b'),
+                ('a', 'b', 'c', 'e')}
+        assert [] == errors(table_batch)
 
-            assert {} == table_batch['streamed_schema']['schema']['properties']
 
-            assert 5 == len(table_batch['records'])
+def _get_table_batch_with_path(table_batches, path):
+    for table_batch in table_batches:
+        if path == table_batch['streamed_schema']['path']:
+            return path
```

Comment:
Here we need to return `table_batch` instead of `path` otherwise we fail the tests on trying to access 'schema' etc. I know you meant do that since the function name is `_get_table_batch_with_path` :).

Reply:
🤦‍♂️ this has been a hilariously tough thing for me to do as I don't actually _have_ the working code etc.

Reply:
Yeah I can imagine that 🙂. Anyway, great work! 🚀